component/file_cache/file_cache.go

```diff
@@ -691,69 +691,88 @@ func (fc *FileCache) DeleteFile(options internal.DeleteFileOptions) error {
 	return nil
 }
 
-// OpenFile: Makes the file available in the local cache for further file operations.
-func (fc *FileCache) OpenFile(options internal.OpenFileOptions) (*handlemap.Handle, error) {
-	log.Trace("FileCache::OpenFile : name=%s, flags=%d, mode=%s", options.Name, options.Flags, options.Mode)
+func (fc *FileCache) DownloadFile(options internal.DownloadFileOptions) (*handlemap.Handle, error) {
+
+	//extract flag and mode out of the value from handle
+	var flag int
+	var fMode fs.FileMode
+
+	flagMode, found := options.Handle.GetValue("fileFlagMode")
+	if found {
+		flagModeStruct, ok := flagMode.(struct {
```

Comment:
`flags` and `mode` are the information needed to open the file once it's downloaded.
If the name of the field, struct, and variable they're in reflected their purpose, it'd probably be clearer. Maybe `openFileOptions`?